src/zabapgit_transport.prog.abap

```diff
@@ -203,5 +183,140 @@ CLASS lcl_transport IMPLEMENTATION.
+  PRIVATE SECTION.
+
+    METHODS create_new_branch
+      IMPORTING
+        io_repository  TYPE REF TO lcl_repo_online
+        iv_branch_name TYPE string
+      RAISING
+        lcx_exception.
+    METHODS generate_commit_message
+      IMPORTING
+        is_transport_to_branch TYPE ty_transport_to_branch
+      RETURNING
+        VALUE(rs_comment)      TYPE ty_comment.
+    METHODS stage_transport_objects
+      IMPORTING
+        it_transport_objects TYPE scts_tadir
+        io_stage             TYPE REF TO lcl_stage
+        is_stage_objects     TYPE ty_stage_files
+        it_object_statuses   TYPE ty_results_tt
+      RAISING
+        lcx_exception.
+ENDCLASS.
+
+CLASS lcl_transport_to_branch IMPLEMENTATION.
+
+  METHOD create.
+    DATA:
+      ls_transport_object TYPE LINE OF scts_tadir,
+      lt_items            TYPE ty_files_item_tt,
+      ls_local_file       TYPE LINE OF ty_files_item_tt,
+      ls_remote_file      TYPE LINE OF ty_files_tt,
+      ls_item             TYPE string,
+      lv_branch_name      TYPE string,
+      ls_comment          TYPE ty_comment,
+      lo_stage            TYPE REF TO lcl_stage,
+      ls_stage_objects    TYPE ty_stage_files,
+      lt_object_statuses  TYPE ty_results_tt,
+      ls_object_status    TYPE LINE OF ty_results_tt.
+
+    lv_branch_name = lcl_git_branch_list=>complete_heads_branch_name(
+        lcl_git_branch_list=>normalize_branch_name( is_transport_to_branch-branch_name ) ).
+
+    create_new_branch(
+      io_repository  = io_repository
+      iv_branch_name = lv_branch_name ).
+
+    CREATE OBJECT lo_stage
+      EXPORTING
+        iv_branch_name = lv_branch_name
+        iv_branch_sha1 = io_repository->get_sha1_remote( ).
+
+    ls_stage_objects = lcl_stage_logic=>get( io_repository ).
+
+    lt_object_statuses = io_repository->status( ).
+
+    stage_transport_objects(
+       it_transport_objects = it_transport_objects
+       io_stage             = lo_stage
+       is_stage_objects     = ls_stage_objects
+       it_object_statuses   = lt_object_statuses ).
+
+    ls_comment = generate_commit_message( is_transport_to_branch ).
+
+    io_repository->push( is_comment = ls_comment
+                         io_stage   = lo_stage ).
+  ENDMETHOD.
+
+  METHOD create_new_branch.
+    ASSERT iv_branch_name CP 'refs/heads/+*'.
+    TRY.
+        lcl_git_porcelain=>create_branch(
+          io_repo = io_repository
+          iv_name = iv_branch_name
+          iv_from = io_repository->get_sha1_local( ) ).
+
+        io_repository->set_branch_name( iv_branch_name ).
+      CATCH lcx_exception.
+        lcx_exception=>raise( 'Error when creating new branch').
+    ENDTRY.
+  ENDMETHOD.
+
+  METHOD generate_commit_message.
+    rs_comment-committer-name  = sy-uname.
+    rs_comment-committer-email = |{ rs_comment-committer-name }@localhost|.
+    rs_comment-comment         = is_transport_to_branch-commit_text.
+  ENDMETHOD.
+
+
+  METHOD stage_transport_objects.
```

Comment:
In a next pull request, I'll extract this method to a class and put under test. This is the core logic...